src/SessionMonitor/index.js

```diff
@@ -35,25 +36,62 @@ export class SessionMonitor {
     this.mostRecentActivityTimestamp = Date.now();
   }
 
+  isUserOnPage(pageName) {
+    const paths = window.location.href.split('/').filter(x => x !== 'dev.html');
+    return paths[paths.length - 1] === pageName;
+  }
+
   updateSession() {
-    // If user has been inactive for Y min
-    if (Date.now() - this.mostRecentActivityTimestamp >= this.inactiveTimeLimit) {
-      this.logoutUser();
-    } else if (Date.now() - this.mostRecentActivityTimestamp < this.inactiveTimeLimit) {
-      this.refreshSession();
-    } else {
-      console.log('Error calculating inactive time');
+    if (this.isUserOnPage('login')) {
+      return Promise.resolve(0);
     }
+
+    const timeSinceLastActivity = Date.now() - this.mostRecentActivityTimestamp;
+    // If user has been inactive for Y min, and they are not in a workspace
+    if (timeSinceLastActivity >= this.inactiveTimeLimit
+        && !this.isUserOnPage('workspace')
+        && logoutInactiveUsers) {
+      // Allow Fence to log out the user. If we don't refresh, Fence will mark them as inactive.
+      this.notifyUserIfTheyAreNotLoggedIn();
+      return Promise.resolve(0);
+    }
+
+    return this.refreshSession();
   }
 
   refreshSession() {
+    if (this.isUserOnPage('login')) {
+      return Promise.resolve(0);
+    }
     // hitting Fence endpoint refreshes token
-    fetch(userapiPath);
+    const self = this;
```

Comment:
I think ES6's arrow function will take `this` as the one from outer scope, so you don't need to create an extra `self` here